rosidl_generator_py/resource/_msg_support.c.em

```diff
@@ -370,6 +405,28 @@ nested_type = '__'.join(type_.namespaced_name())
     }
     rosidl_generator_c__String__assign(&ros_message->@(member.name), PyBytes_AS_STRING(encoded_field));
     Py_DECREF(encoded_field);
+@[  elif isinstance(member.type, AbstractWString)]@
+    assert(PyUnicode_Check(field));
+    // the returned string starts with a BOM mark
+    PyObject * encoded_field = PyUnicode_AsUTF16String(field);
+    if (!encoded_field) {
+      Py_DECREF(field);
+      return false;
+    }
+    char * buffer;
+    Py_ssize_t length;
+    int rc = PyBytes_AsStringAndSize(encoded_field, &buffer, &length);
+    if (rc) {
+      Py_DECREF(field);
+      return false;
+    }
+    // use offset of 2 to skip BOM mark
+    bool succeeded = rosidl_generator_c__U16String__assignn_from_char(&ros_message->@(member.name), buffer + 2, length - 2);
+    if (!succeeded) {
+      Py_DECREF(field);
+      return false;
+    }
+    Py_DECREF(encoded_field);
```

Comment:
We should ensure this is decref'd before or in the two preceding failure cases.

Reply:
Good catch: see 03a891edf95220079d16febdf9ad3175647f0958.

Reply:
You moved one of them, but this one should still be moved.

Reply:
And here it is: 1edd2365b6af17e46abdef5951aa154eb78fb7a8.